From 9a7dbc7047da0e6ca52ff415dcd448b7333a1798 Mon Sep 17 00:00:00 2001
From: Olivier BIDEAU <olivier.bideau@st.com>
Date: Tue, 16 Jan 2024 10:22:43 +0100
Subject: [PATCH 0505/1695] drivers: stm32_rng: keep random generation on when
 optee is in sysram

This patch allows to maintain the random generation
when suspending the platform with optee in sysram.

In this case the RNG is needed to allow context encryption
before storing it in DDR.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git be7f633d0de5e72c4803ad8477b7ef621127ef6c drivers: stm32_rng: keep random generation on when optee is in sysram]
Signed-off-by: Olivier BIDEAU <olivier.bideau@st.com>
Change-Id: Ie8f8de1a261eccad8eaa96c9990d54968188d168
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/351510
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/stm32_rng.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/drivers/stm32_rng.c b/core/drivers/stm32_rng.c
index b00e09739..2c77fea81 100644
--- a/core/drivers/stm32_rng.c
+++ b/core/drivers/stm32_rng.c
@@ -552,7 +552,9 @@ static TEE_Result stm32_rng_pm_suspend(void)
 		if ((io_read32(rng_base + RNG_CR) & RNG_CR_CONDRST))
 			panic();
 	} else {
+#ifndef CFG_STM32MP1_OPTEE_IN_SYSRAM
 		io_clrbits32(rng_base + RNG_CR, RNG_CR_RNGEN);
+#endif
 	}
 
 	return TEE_SUCCESS;
-- 
2.47.2

